micropip/package_index.py

```diff
@@ -39,7 +39,7 @@ class ProjectInfo:
     # List of releases available for the package, sorted in ascending order by version.
     # For each version, list of wheels compatible with the current platform are stored.
     # If no such wheel is available, the list is empty.
-    releases: dict[Version, Generator[WheelInfo, None, None]]
+    releases: dict[CachedVersion, Generator[WheelInfo, None, None]]
```

Comment:
CachedVersion is a child class of the original Version, so I think the type hints do not need to change.